spec/interactor/context_spec.rb

```diff
@@ -57,8 +57,7 @@ module Interactor
     end
 
     describe "#fail!" do
-      let(:context) { Context.build(foo: "bar") }
-      let(:error_context) { Context.build }
+      let(:context) { Context.build(foo: "bar", user: {password: "12345678", password_confirmation: '12345678'}) }
```

Comment:
One important `rspec` construct is to keep `let()` definitions as generic as possible. That is, define the minimal possible data that can be used across all tests, and let each individual test build out the data models it needs . The `password` and `password_confirmation` keys here don't apply to all scenarios being tested, they only apply to a handful of tests that will verify the redaction logic. 

Adding this here at minimum adds weight to the definition and creates objects that take longer to initialize. The latter is a minor point for small gems like this, but check out [this mess in the callrail app](https://github.com/callrail/callrail/blob/master/engines/abacus/spec/queries/abacus/aggregators/pivot_tables/by_interval_spec.rb#L21..L44) (i just picked one example, we do that in a _lot_ of places.) We define every data scenario that each test needs at once instead of letting each test build it's own `Call` and `Company` objects. [Here](https://github.com/callrail/callrail/pull/21053/files#r508533824) is another callrail example of that being fixed. 

In your case you're building a feature (redaction) on top of existing functionality. That is, none of the existing tests should break or need to be modified, you're just strictly adding new functionality to test a new specific case.

I'd look at re-working the below so that it doesn't modify existing tests. You can add a new group of tests with a `describe` or `context` block if needed (they're the same thing)

```ruby
context('context has one or more redactable keys') do
  it('redacts the redactable keys') do
    #...
  end

  it('preserves the original keys') do
    #...
  end

  it('...') do
    #...
  end
end
```

Reply:
Yeah, this makes a lot of sense.  I will update accordingly.  It seems to fix my issue with the spec too!